app/src/test/java/com/github/se/eventradar/MockEventRepositoryUnitTest.kt

```diff
@@ -0,0 +1,132 @@
+package com.github.se.eventradar
+
+import com.github.se.eventradar.model.Location
+import com.github.se.eventradar.model.Resource
+import com.github.se.eventradar.model.event.Event
+import com.github.se.eventradar.model.event.EventCategory
+import com.github.se.eventradar.model.event.EventTicket
+import com.github.se.eventradar.model.repository.event.IEventRepository
+import com.github.se.eventradar.model.repository.event.MockEventRepository
+import java.time.LocalDateTime
+import kotlinx.coroutines.test.runTest
+import org.junit.Before
+import org.junit.Test
+
+class MockEventRepositoryUnitTest {
+  private lateinit var eventRepository: IEventRepository
```

Comment:
Ok that's nice! Two things to keep in mind:
- be careful with concurrency => tests will execute in parallel
- with this approach, you are usnig **manual dependency injection** (i.e. you are saying "I create a repository of type IEventRepository" and test it). In other words, you are using Hilt (automated injection) for the app, and manual injection for the tests. Since you defined the `@Binds` in the modules above, you should be able to write the following

```kotlin
@HiltAndroidTest
@RunWith(AndroidJUnit4::class)
class MockEventRepositoryUnitTest {

    @get:Rule
    var hiltRule = HiltAndroidRule(this)

    @Before
    fun setUp() {
        hiltRule.inject() // inject dependencies using the module for your tests
    }

    // write tests below
}
```

Reply:
I don't believe I am able to do this in a unit test (under `test`, not `androidTest`. Since I am only explicitly testing the mock database, I think it is fine to not use the injection logic here.